carsus/io/output/base.py

```diff
@@ -33,21 +36,54 @@ class TARDISAtomData:
         Dump all attributes into an HDF5 file
     """
 
-    def __init__(self, gfall_reader, ionization_energies, ions,
+    def __init__(self,
+                 atomic_weights,
+                 ionization_energies,
+                 gfall_reader,
+                 zeta_data,
+                 chianti_reader=None,
                  lines_loggf_threshold=-3,
                  levels_metastable_loggf_threshold=-3):
 
+        # TODO: pass these params to the function as `gfall_params`
         self.levels_lines_param = {
             "levels_metastable_loggf_threshold":
             levels_metastable_loggf_threshold,
             "lines_loggf_threshold": lines_loggf_threshold
         }
 
-        self.ions = parse_selected_species(ions)
+        self.atomic_weights = atomic_weights
+        self.ionization_energies = ionization_energies
+        self.ground_levels = ionization_energies.get_ground_levels()
+
+        gfall_ions = gfall_reader.levels.index.tolist()
+        # Remove last element from tuple (MultiIndex has 3 elements)
+        gfall_ions = [x[:-1] for x in gfall_ions]
+        # Keep unique tuples, list and sort them
+        gfall_ions = sorted(list(set(gfall_ions)))
         self.gfall_reader = gfall_reader
+        self.gfall_ions = gfall_ions
 
-        self.ionization_energies = ionization_energies.base
-        self.ground_levels = ionization_energies.get_ground_levels()
+        self.zeta_data = zeta_data
+
+        # TODO: priorities should not be managed by the `init` method.
+        self.chianti_reader = chianti_reader
+        if chianti_reader is not None:
+            chianti_lvls = chianti_reader.levels.reset_index()
+            chianti_lvls = chianti_lvls.set_index(
+                ['atomic_number', 'ion_charge', 'priority'])
+
+            mask = chianti_lvls.index.get_level_values(
+                'priority') > self.gfall_reader.priority
```

Comment:
I think in the future we should make sure that the priority values are unique (e.g. chianti and gfall should not both have priority 10). What source is used should then be decided solely on the priority value (to be more transparent). For now, it is probably fine.

Reply:
I agree, we have to decide which one has higher priority (and even more so if we include CMFGEN)

Reply:
The priority implementation needs more discussion, for sure.